python/lsst/meas/extensions/multiprofit/fit_coadd_multiband.py

```diff
@@ -152,142 +206,202 @@ class MultiProFitSourceTask(CatalogSourceFitterABC, fitMB.CoaddMultibandFitSubTa
+        centroid_pixel_offset: float = 0,
+        **kwargs,
     ):
-        comps = model.sources[0].components
-        sig_x = math.sqrt(source["base_SdssShape_xx"])
-        sig_y = math.sqrt(source["base_SdssShape_yy"])
-        # There is a sign convention difference
-        rho = np.clip(-source["base_SdssShape_xy"] / (sig_x * sig_y), -0.5, 0.5)
+        if values_init is None:
+            values_init = {}
+        set_flux_limits = kwargs.pop("set_flux_limits") if "set_flux_limits" in kwargs else True
+        if kwargs:
+            raise ValueError(f"Unexpected {kwargs=}")
+        sig_x = math.sqrt(source["slot_Shape_xx"])
+        sig_y = math.sqrt(source["slot_Shape_yy"])
+        # TODO: Verify if there's a sign difference here
+        rho = np.clip(source["slot_Shape_xy"] / (sig_x * sig_y), -0.5, 0.5)
         if not np.isfinite(rho):
             sig_x, sig_y, rho = 0.5, 0.5, 0
-        if not source["base_SdssShape_flag"]:
-            flux = source["base_SdssShape_instFlux"]
-        else:
-            flux = source["base_GaussianFlux_instFlux"]
-            if not (flux > 0):
-                flux = source["base_PsfFlux_instFlux"]
-                if not (flux > 0):
-                    flux = 1
-        n_psfs = self.config.n_pointsources
-        n_extended = len(self.config.sersics)
-        observation = model.data[0]
-        x_max = float(observation.image.n_cols)
-        y_max = float(observation.image.n_rows)
-        limits_x.max = x_max
-        limits_y.max = y_max
+
+        # Make restrictive centroid limits (intersection, not union)
+        x_min, y_min, x_max, y_max = -np.Inf, -np.Inf, np.Inf, np.Inf
+
+        fluxes_init = []
+        fluxes_limits = []
+
+        n_observations = len(model.data)
+        n_components = len(model.sources[0].components)
+
+        for idx_obs, observation in enumerate(model.data):
+            coordsys = observation.image.coordsys
+            catexp = catexps[idx_obs]
+            band = catexp.band
+
+            x_min = max(x_min, coordsys.x_min)
+            y_min = max(y_min, coordsys.y_min)
+            x_max = min(x_max, coordsys.x_min + float(observation.image.n_cols))
+            y_max = min(y_max, coordsys.y_min + float(observation.image.n_rows))
+
+            flux_total = np.nansum(observation.image.data[observation.mask_inv.data])
+
+            column_ref = f"merge_measurement_{band}"
+            if column_ref in source.schema.getNames() and source[column_ref]:
+                row = source
+            else:
+                row = catexp.catalog.find(source["id"])
+
+            if not row["base_SdssShape_flag"]:
+                flux_init = row["base_SdssShape_instFlux"]
+            else:
+                flux_init = row["slot_GaussianFlux_instFlux"]
+                if not (flux_init > 0):
+                    flux_init = row["slot_PsfFlux_instFlux"]
+
+            photoCalib = catexp.exposure.photoCalib
```

Comment:
Make this photo_calib. I think that is the usual way to do this, even though the name in the other package uses camel case.